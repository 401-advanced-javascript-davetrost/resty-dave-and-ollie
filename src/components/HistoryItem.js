import React from 'react';
import PropTypes from 'prop-types';

export const HistoryItem = ({ _id, url, method, handleHistorySelect }) => {
  return (
    <div onClick={event => handleHistorySelect(event, _id)}>
      <h3>{method}</h3>
      <p>{url}</p>
    </div>
  );
};

HistoryItem.propTypes = {
  _id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string,
  handleHistorySelect: PropTypes.func.isRequired
};
