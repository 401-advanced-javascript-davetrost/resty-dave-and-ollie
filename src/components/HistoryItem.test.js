import React from 'react';
import { shallow } from 'enzyme';
import { HistoryItem } from './HistoryItem';

describe('HistoryItem component', () => {
  it('renders a history item', () => {
    const wrapper = shallow(<HistoryItem 
      _id="abc123"
      url="heyarnold.com/api/characters"
      method="get"
      handleHistorySelect={() => true}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a history item with the body included', () => {
    const wrapper = shallow(<HistoryItem 
      _id="xyzabc"
      url="random-api.com/api/characters"
      method="post"
      body={`{
        name: 'Garfield',
        color: 'Orange with black stripes'
      }
      `}
      handleHistorySelect={() => true}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
