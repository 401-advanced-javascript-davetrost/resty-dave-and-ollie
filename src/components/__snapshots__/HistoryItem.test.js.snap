// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem component renders a history item 1`] = `
<div
  onClick={[Function]}
>
  <h3>
    get
  </h3>
  <p>
    heyarnold.com/api/characters
  </p>
</div>
`;

exports[`HistoryItem component renders a history item with the body included 1`] = `
<div
  onClick={[Function]}
>
  <h3>
    post
  </h3>
  <p>
    random-api.com/api/characters
  </p>
</div>
`;
